Visual-Riot/plaeen
# Page/calendar

**Issue:** link to git issue if available

**Designs:** link to design if available

**This PR:**

- User calendar is now done. When loading, it displays skeleton. Then it loads data from prisma. Slots can be changed and are stored temporarily in localStorage. Upon clicking reset it will go back to prisma last saved state. When clicked on SAVE it will take data from localStorage and send it to prisma. It will also reset localstorage, so next changes can be made.

**Steps to test:**

1. Open localhost:3000/calendar and check if SKELETON is loaded correctly
2. If opening first time, not data should be displayed (all slots should be blank)
3. Change states of some slots and press Save. Refresh page to see if they are displayed correctly.
4. Test RESET button to see if it resets to last prisma state after changing some states of slots without saving.

**Checklist:**

- [] Code changes should not introduce new errors or warnings
- [] Skeleton should load when opening page and appear before the data from prisma is loaded
- [] Reset Button should reset localStorage and reverse unsaved changes
- [] Save button should save changed data to prisma and this should be loaded after refreshing page
